TrigScint/include/TrigScint/Firmware/objdef.h

```diff
@@ -62,55 +66,70 @@ inline void cpyHit(Hit& c1, Hit& c2) {
 }
 
 struct Cluster {
-  Hit Seed;
-  Hit Sec;
-  ap_int<11> Cent;
+  Hit Seed{};
+  Hit Sec{};
+  ap_int<12> Cent{};
   // int nhits, mID, SeedID;
   // float CentX, CentY, CentZ, Amp, Time, TrigTime;
 };
+
 inline void clearClus(Cluster& c) {
   clearHit(c.Seed);
   clearHit(c.Sec);
-  c.Cent = (ap_int<11>)(0);  // clearHit(c.For);
+  c.Cent = (ap_int<12>)(0);  // clearHit(c.For);
 }
+
 inline void calcCent(Cluster& c) {
-  if (c.Seed.Amp > 0) {
-    c.Cent = (ap_int<12>)(10. *
-                          ((float)(c.Seed.Amp * c.Seed.bID +
-                                   c.Sec.Amp * c.Sec.bID)) /
-                          ((float)(c.Seed.Amp + c.Sec.Amp)));
-  } else {
+  // Check if Seed and Sec amplitudes are valid
+  if (c.Seed.Amp <= 0 || c.Sec.Amp <= 0) {
+    c.Cent = (ap_int<12>)(0);
+    return;
+  }
+
+  if (c.Seed.bID < 0 || c.Sec.bID < 0) {
     c.Cent = (ap_int<12>)(0);
+    return;
   }
+
+  // Perform the centroid calculation if all checks passed
+  c.Cent =
+      (ap_int<12>)(10.0f *
+                   ((float)(c.Seed.Amp * c.Seed.bID + c.Sec.Amp * c.Sec.bID)) /
+                   ((float)(c.Seed.Amp + c.Sec.Amp)));
 }
+
 inline void cpyCluster(Cluster& c1, Cluster& c2) {
   cpyHit(c1.Seed, c2.Seed);
   cpyHit(c1.Sec, c2.Sec);
 }
 
 struct Track {
-  Cluster Pad1;
-  Cluster Pad2;
-  Cluster Pad3;
-  ap_int<12> resid;
+  Cluster Pad1{};
+  Cluster Pad2{};
+  Cluster Pad3{};
+  ap_int<12> resid{};
 };
+
 inline void clearTrack(Track& c) {
   clearClus(c.Pad1);
   clearClus(c.Pad2);
   clearClus(c.Pad3);
   c.resid = 5000;
 }
+
 inline ap_int<12> calcTCent(Track& c) {
   calcCent(c.Pad1);
   calcCent(c.Pad2);
   calcCent(c.Pad3);
+
   float one = (float)c.Pad1.Cent;
   float two = (float)c.Pad2.Cent;
   float three = (float)c.Pad3.Cent;
   float mean = (one + two + three) / 3.0;
-  ap_int<12> Cent = (ap_int<10>)((int)(mean));
+  ap_int<12> Cent = (ap_int<12>)((int)(mean));

```

Comment:
I really think this should be as I change it, but please check

Reply:
This will not significantly affect the firmware part of this equation. If it agrees still on track positions, it will also work. I will clone this branch and check gimme a sec.